### ISSUE TITLE
the app does not follow the deeplink

### ISSUE THREAD

**Issue by stebaia** [suleymangunes/deeplinking-auto_route#4]
hi! i am using auto_route in my app, i followed your guide by setting auto_route as you did.

I'm testing on ios, but when I open a link (any) that has my site as url base, the app opens, but doesn't follow the deeplink.

I've also tried putting print and break points inside the deepLinkBuilder, but nothing is read.

The app just opens

**Comment by workspace-cs** [comment]
Hello, unfortunately I was only able to test the application on android.

Can you test your project on android?

I think we can better approach the solution if we understand whether the problem is in the code part or in the ios part.

**Comment by kahakai** [comment]
> if we understand whether the problem is in the code part or in the ios part.

Yes, the problem is in the code part. You haven't opted in to deep linking for iOS as per [documentation](https://docs.flutter.dev/cookbook/navigation/set-up-universal-links#adjust-ios-build-settings).

> If you use a Flutter version earlier than 3.27, you need to manually opt in to deep linking by adding the key and value pair `FlutterDeepLinkingEnabled` and `YES` to `Info.plist`.

Then you need to add the associated domains (doesn't work for Personal account, because "Personal development teams don't support the Associated Domains capability.", you have to have a paid Apple (Individual / Organization) Developer Account):
https://docs.flutter.dev/cookbook/navigation/set-up-universal-links#add-associated-domains

Last, you need to create and host `apple-app-site-association` JSON file:
https://docs.flutter.dev/cookbook/navigation/set-up-universal-links#create-and-host-apple-app-site-association-json-file
